lib/ui/settings/data/export_widget.dart

```diff
@@ -122,32 +123,53 @@ Future<void> _showExportWarningDialog(BuildContext context) async {
 }
 
 Future<void> _exportCodes(BuildContext context, String fileContent) async {
-  final _codeFile = File(
-    Configuration.instance.getTempDirectory() + "ente-authenticator-codes.txt",
+  final codeFile = File(
+    "${Configuration.instance.getTempDirectory()}ente-authenticator-codes.txt",
   );
   final hasAuthenticated = await LocalAuthenticationService.instance
       .requestLocalAuthentication(context, context.l10n.authToExportCodes);
   if (!hasAuthenticated) {
     return;
   }
-  if (_codeFile.existsSync()) {
-    await _codeFile.delete();
-  }
-  _codeFile.writeAsStringSync(fileContent);
-  final Size size = MediaQuery.of(context).size;
-  await Share.shareFiles([_codeFile.path], sharePositionOrigin: Rect.fromLTWH(0, 0, size.width, size.height / 2),);
-  Future.delayed(const Duration(seconds: 15), () async {
-    if (_codeFile.existsSync()) {
-      _codeFile.deleteSync();
-    }
-  });
+  Future.delayed(
+    const Duration(milliseconds: 1200),
+    () async => await shareDialog(
```

Comment:
Curious, why the delay here?

Reply:
It is because the dialog takes around 1 seconds to pop, so if we show the dialog before that then on pop will be called on this shareDialog.

I will try to link the code where the delay is there but you can actually see the delay too in this export code dialog (don't know why the delay was there before).